symphony/bdk/core/service/presence/presence_service.py

```diff
@@ -0,0 +1,176 @@
+        :return: Presence info of the looked up user.
+        `Get All Presence <https://developers.symphony.com/restapi/reference#get-all-presence>`_
+        """
+        return await self._presence_api.v2_users_presence_get(session_token=await self._auth_session.session_token,
+                                                              last_user_id=last_user_id,
+                                                              limit=limit)
+
+    async def get_user_presence(self, user_id: int, local: bool) -> V2Presence:
+        """ Get the presence info of a specified user.
+
+        :param user_id: User Id
+        :param local: If true then Perform a local query and set the presence to OFFLINE for  users who are not local to
+        the calling user’s pod. If false or absent then query the presence of all local  and external users who are
+        connected to the calling user.
+
+        :return: Presence info of the looked up user.
+        `Get User Presence <https://developers.symphony.com/restapi/reference#user-presence-v3`_
+        """
+        return await self._presence_api.v3_user_uid_presence_get(uid=user_id,
+                                                                 session_token=await self._auth_session.session_token,
+                                                                 local=local)
+
+    async def external_presence_interest(self, user_ids: List[int]):
+        """ Register interest in a list of external users to get their presence info.
+
+        :param user_ids: List of user ids to be registered.
+        `External Presence Interest <https://developers.symphony.com/restapi/reference#register-user-presence-interest`_
+        """
+        await self._presence_api.v1_user_presence_register_post(session_token=await self._auth_session.session_token,
+                                                                uid_list=user_ids)
+
+    async def set_presence(self, status: PresenceStatus, soft: bool) -> V2Presence:
+        """ Set the presence info of the calling user.
+
+        :param status: The new presence state for the user.
+        Possible values are AVAILABLE, BUSY, AWAY, ON_THE_PHONE, BE_RIGHT_BACK, IN_A_MEETING, OUT_OF_OFFICE, OFF_WORK.
+
+        :param soft: If true, the user's current status is taken into consideration. If the user is currently OFFLINE,
+        the user's presence will still be OFFLINE, but the new presence will take effect when the user comes online. If
+        the user is currently online, the user's activity state will be applied to the presence if applicable. (e.g. if
+        you are setting their presence to AVAILABLE, but the user is currently idle, their status will be represented as
+        AWAY)
+        :return: Presence info of the calling user.
+        `Set Presence <https://developers.symphony.com/restapi/reference#set-presence"`_
+        """
+        presence_status: V2PresenceStatus = V2PresenceStatus(category=status.value)
```

Comment:
Regarding my comment on PresenceStatus enum, you could use `status.name` instead.